src/Adapter/Product/CommandHandler/UpdateCustomizationFieldHandler.php

```diff
@@ -28,70 +28,88 @@
  */
 final class UpdateCustomizationFieldHandler extends AbstractCustomizationFieldHandler implements UpdateCustomizationFieldHandlerInterface
 {
+    /**
+     * @var ProductProvider
+     */
+    private $productProvider;
+
+    /**
+     * @var ProductUpdater
+     */
+    private $productUpdater;
+
+    /**
+     * @var CustomizationFieldUpdater
+     */
+    private $customizationFieldUpdater;
+
+    /**
+     * @param ProductProvider $productProvider
+     * @param ProductUpdater $productUpdater
+     * @param CustomizationFieldUpdater $customizationFieldUpdater
+     */
+    public function __construct(
+        ProductProvider $productProvider,
+        ProductUpdater $productUpdater,
+        CustomizationFieldUpdater $customizationFieldUpdater
+    ) {
+        $this->productProvider = $productProvider;
+        $this->productUpdater = $productUpdater;
+        $this->customizationFieldUpdater = $customizationFieldUpdater;
+    }
+
     /**
      * {@inheritdoc}
      */
     public function handle(UpdateCustomizationFieldCommand $command): void
     {
         $customizationField = $this->getCustomizationField($command->getCustomizationFieldId());
-        $this->fillEntityWithCommandData($command, $customizationField);
-        $this->validateCustomizationFieldName($customizationField);
+        $this->customizationFieldUpdater->update($customizationField, $this->formatPropertiesForUpdate($command));
 
-        try {
-            if (false === $customizationField->update()) {
-                throw new CannotUpdateCustomizationFieldException(sprintf(
-                    'Failed to update customization field #%s',
-                    $customizationField->id
-                ));
-            }
-        } catch (PrestaShopException $e) {
-            throw new CustomizationFieldException(sprintf(
-                'Error occurred when trying to update customization field #%d',
-                $customizationField->id
-            ));
-        }
-
-        $product = $this->getProduct(new ProductId((int) $customizationField->id_product));
-        $this->refreshProductCustomizability($product);
-        $this->refreshCustomizationFieldsCount($product);
-        $this->performUpdate($product, CannotUpdateProductException::FAILED_UPDATE_CUSTOMIZATION_FIELDS);
+        $product = $this->productProvider->get(new ProductId((int) $customizationField->id_product));
+        $this->productUpdater->refreshProductCustomizabilityFields($product);
+        $this->productUpdater->update($product, CannotUpdateProductException::FAILED_UPDATE_CUSTOMIZATION_FIELDS);
     }
 
     /**
      * @param UpdateCustomizationFieldCommand $command
-     * @param CustomizationField $field
+     *
+     * @return array
      */
-    private function fillEntityWithCommandData(UpdateCustomizationFieldCommand $command, CustomizationField $field): void
+    private function formatPropertiesForUpdate(UpdateCustomizationFieldCommand $command): array
```

Comment:
I get that you didn't like the array formatting that much, but when you see the diff here (and most handlers will look this way I think) it's actually not that different from populating individual fields one by one